### DISCUSSION TITLE
Docker test configuration file

### DISCUSSION THREAD

**Discussion by krzysztofMajchrzak-GIT** [whittlem/pycryptobot#799] [Q&A]
I was trying to run a few tests using configuration files but can't get it to work. So it seems like it is impossible to run bot with config file without providing API key. How to run tests using json files?

My command (I am locally build docker image):
`sudo docker run -it -v ./001.json:/app/config.json --rm --user pycryptobot pycryptobot/pycryptobot`

My Json config file:
```
"binance": {
    "config": {
        "exchange": binance,
        "market": ETHUSDT,
        "autorestart": 1,
        "base_currency": "ETH",
        "disablebuynearhigh": 1,
        "granularity": "15m",
        "nobuynearhighpcnt": 1,
        "nosellmaxpcnt": 3,
        "nosellminpcnt": -9,
        "quote_currency": "USDT",
        "sellatresistance": 1,
        "selllowerpcnt": -12,
        "trailingbuypcnt": 1,
        "trailingstoploss": -1,
        "trailingstoplosstrigger": 3,
        "sim": fast,
        "simstartingdate": "2023-01-01"
    }
}
```

And the error i get:

````
Traceback (most recent call last):
  File "/app/models/config/binance_parser.py", line 105, in parser
    with open(app.api_key_file, "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'binance.key'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/pycryptobot.py", line 21, in <module>
    main()
  File "/app/pycryptobot.py", line 12, in main
    app = PyCryptoBot()
  File "/app/controllers/PyCryptoBot.py", line 78, in __init__
    super(PyCryptoBot, self).__init__(filename=self.config_file, exchange=exchange)
  File "/app/models/BotConfig.py", line 180, in __init__
    self.read_config()
  File "/app/models/BotConfig.py", line 230, in read_config
    binanceConfigParser(self, self.config[self.exchange.value], self.cli_args)
  File "/app/models/config/binance_parser.py", line 111, in parser
    raise RuntimeError(f"Unable to read {app.api_key_file}")
RuntimeError: Unable to read binance.key
````
One more thing i cant use properly is logging and graphs. using this command:
` sudo docker run -it --rm -v ./graphs:/app/graphs -v ./testlog.log:/app/pycryptobot.log -v ./testtrade.csv:/app/trades.csv --user pycryptobot pycryptobot/pycryptobot bash --exchange binance --market ETHUSDT --granularity 15m --sim fast --autorestart 1 --base_currency "ETH" --quote_currency "USDT" --disablebuynearhigh 1 --nobuynearhighpcnt 1  --sellatresistance 1 --selllowerpcnt -5 --trailingbuypcnt 1 --trailingstoploss -1 --trailingstoplosstrigger 3 --simstartdate "2022-01-01" --simenddate "2022-02-01" --predictions 1 --preventloss 1 --logs 0 --tradetracker 1 --logbuysellinjson 1`

i get logfile that is a copy of terminal output, no graphs and no json files. I want to have pure json file with only transactions not all logs.

Thank you in advance.

**Comment by r-brown** [comment]
You need to map your `binance.key` in the container as well
just add this to the CLI:
`v ./binance.key:/app/binance.key`

**Comment by JacekZubielik** [comment]
If you want to specify the start date of the simulation sample you should use `simstartdate` instead of `simstartingdate`.
